files/image_config/procdockerstats/procdockerstats

```diff
@@ -0,0 +1,122 @@
+    
+    def run_command(self, cmd):
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+	(stdout, stderr) = proc.communicate()
+	
+	if proc.returncode != 0:
+	    log_error("Error running command '{}'".format(cmd))
+	else:
+            return stdout
+
+    def format_docker_cmd_output(self, cmdout):
+        lines = re.split("\n", cmdout)
+        keys = re.split("   +", lines[0])
+        x = len(lines)
+        dict1 = dict()
+        dict_list = []
+        for i in range(1,x):
+            values1 = re.split("   +", lines[i])
+            dict1 =  dict(zip(keys, values1))
+            dict_list.append(dict1)
+        return dict_list
+    
+    def update_dockerstats_command(self):        
+        data = self.run_command("docker stats --no-stream -a")
+        dockerdata = self.format_docker_cmd_output(data)
+
+        value = ""
+        # wipe out all data before updating with new values
+        self.state_db.delete_all_by_pattern('STATE_DB', 'Docker_Stats|*')
+        for index in range(len(dockerdata)):
+            row = dockerdata[index]
+            cid = row.get('CONTAINER ID')
+            if cid:
+                value = 'Docker_Stats|' + str(cid)
+                name = row.get('NAME')
+                self.update_state_db(value, 'NAME', name)
+                cpu = row.get('CPU %')
+                self.update_state_db(value, 'CPU%', str(cpu))
+                splitcol = row.get('MEM USAGE / LIMIT')
+                memuse = re.split(" / ", str(splitcol))
+                self.update_state_db(value, 'MEM', str(memuse[0]))
+                self.update_state_db(value, 'MEM_LIMIT', str(memuse[1]))
+                mem = row.get('MEM %')
+                self.update_state_db(value, 'MEM %', str(mem))
+                splitcol = row.get('NET I/O')
+                netio = re.split(" / ", str(splitcol))
+                self.update_state_db(value, 'NET_IN', str(netio[0]))
+                self.update_state_db(value, 'NET_OUT', str(netio[1]))
+                splitcol = row.get('BLOCK I/O')
+                blocio = re.split(" / ", str(splitcol))
+                self.update_state_db(value, 'BLOCK_IN', str(blocio[0]))
+                self.update_state_db(value, 'BLOCK_OUT', str(blocio[1]))
+                pids = row.get('PIDS')
+                self.update_state_db(value, 'PIDS', pids)       
+            
+    def update_state_db(self, key1, key2, value2):
+        self.state_db.set('STATE_DB', key1, key2, value2)
+
+    def run(self):
+        self.update_dockerstats_command()
+        datetimeobj = datetime.now()
+        #Adding key to store latest update time.
+        self.update_state_db('Docker_Stats|LastUpdateTime', 'lastupdate', datetimeobj)
+
+# main start
+def main():
+    log_info("process-docker stats aemon starting up..")
+    if not os.getuid() == 0:
+    log_error("Must be root to run process-docker daemon")
+    print "Error: Must be root to run process-docker daemon"
+    sys.exit(1)
```

Comment:
Lines 112-114 should be indented under the `if` statement on line 111.

Reply:
updated

Reply:
Not indented?

Reply:
ah my mistake. I corrected this in iteration 3 but in 4 it got reverted. will fix it